pallets/dca/src/lib.rs

```diff
@@ -199,17 +191,27 @@ pub mod pallet {
 						}
 					}
 				}
+				match schedule.order {
```

Comment:
Note that on line 156 we already accrue the weights

```
				let weight_for_single_execution = Self::get_trade_weight(&schedule.order);
				weight.saturating_accrue(weight_for_single_execution);
```
Maybe this needs to go then?

Reply:
The weight on line 156 is for DCA part, and the second weight is for the router. We use dummy router in the benchmarks so we need to add additional weight.